pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotSortExchangeCopyRule.java

```diff
@@ -42,6 +42,7 @@ public class PinotSortExchangeCopyRule extends RelRule<RelRule.Config> {
 
   public static final PinotSortExchangeCopyRule SORT_EXCHANGE_COPY =
       PinotSortExchangeCopyRule.Config.DEFAULT.toRule();
+  private static final int DEFAULT_SORT_EXCHANGE_COPY_THRESHOLD = 10_000;
```

Comment:
10_000 is a reasonable heuristic until we get a config/hint.

Reply:
is it worth to make this configurable as a query param?

Reply:
probably. but i haven't decided whether to use a hint or use a query param. so for now i will keep this as a default and if we need to support configuration we will do that then.